lib/trino-hive-formats/src/main/java/io/trino/hive/formats/encodings/text/BooleanEncoding.java

```diff
@@ -102,24 +104,38 @@ else if (isFalse(slice, offset, length)) {
     }
 
     @SuppressWarnings("PointlessArithmeticExpression")
-    private static boolean isFalse(Slice slice, int start, int length)
+    private boolean isFalse(Slice slice, int start, int length)
     {
-        return (length == 5) &&
-                (toUpperCase(slice.getByte(start + 0)) == 'F') &&
-                (toUpperCase(slice.getByte(start + 1)) == 'A') &&
-                (toUpperCase(slice.getByte(start + 2)) == 'L') &&
-                (toUpperCase(slice.getByte(start + 3)) == 'S') &&
-                (toUpperCase(slice.getByte(start + 4)) == 'E');
+        if (length == 5) {
+            return (toUpperCase(slice.getByte(start + 0)) == 'F') &&
+                    (toUpperCase(slice.getByte(start + 1)) == 'A') &&
+                    (toUpperCase(slice.getByte(start + 2)) == 'L') &&
+                    (toUpperCase(slice.getByte(start + 3)) == 'S') &&
+                    (toUpperCase(slice.getByte(start + 4)) == 'E');
+        }
+        else if (length == 1 && isExtendedBooleanLiteral) {
+            byte byteValue = slice.getByte(start + 0);
+            return toUpperCase(byteValue) == 'F' ||
+                    byteValue == '0';
+        }
+        return false;
     }
 
     @SuppressWarnings("PointlessArithmeticExpression")
-    private static boolean isTrue(Slice slice, int start, int length)
+    private boolean isTrue(Slice slice, int start, int length)
     {
-        return (length == 4) &&
-                (toUpperCase(slice.getByte(start + 0)) == 'T') &&
-                (toUpperCase(slice.getByte(start + 1)) == 'R') &&
-                (toUpperCase(slice.getByte(start + 2)) == 'U') &&
-                (toUpperCase(slice.getByte(start + 3)) == 'E');
+        if (length == 4) {
+            return (toUpperCase(slice.getByte(start + 0)) == 'T') &&
+                    (toUpperCase(slice.getByte(start + 1)) == 'R') &&
+                    (toUpperCase(slice.getByte(start + 2)) == 'U') &&
+                    (toUpperCase(slice.getByte(start + 3)) == 'E');
+        }
+        else if (length == 1 && isExtendedBooleanLiteral) {
+            byte byteValue = slice.getByte(start + 0);
```

Comment:
redundant `+ 0`

Reply:
addressed the comment. thank you for the review.